packages/unigraph-dev-explorer/src/components/SchemaEditor/SchemaEditor.tsx

```diff
@@ -0,0 +1,78 @@
+import _ from 'lodash';
+import React, { FC, useCallback, useMemo } from 'react';
+import { useList } from 'react-use';
+import Button from '@material-ui/core/Button';
+import IconButton from '@material-ui/core/IconButton';
+
+import SchemaFieldForm from './SchemaFieldForm';
+import { EntityField, makeUnigraphId, makeRefUnigraphId } from './types';
+
+import { makeStyles } from '@material-ui/core/styles';
+import Box from '@material-ui/core/Box';
+import { Add } from '@material-ui/icons';
+
+/**
+ * TODO:
+ *  - ability to add indexes
+ *  - mark field as required
+ */
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    width: 'max-content',
+  }
+}));
+
+interface SchemaEditorProps {
+  onSubmit(schema: any): void;
+}
+
+const SchemaEditor: FC<SchemaEditorProps> = ({ onSubmit }) => {
+  const classes = useStyles();
+  const [fields, { push, updateAt }] = useList<Partial<EntityField>>([{}]);
```

Comment:
Thinking of how we store the entity fields internally - essentially we need to have a trade-off between expressiveness and efficiency. I think making fields into an array makes sense in the current setting to allow better expressiveness but might need further optimization (a naive one would be a hashmap) if we're doing queries over a large amount of data